### ISSUE TITLE
Warning during svelte kit build

### ISSUE THREAD

**Issue by nosovk** [TahaSh/svelte-paginate#23]
If you will try to use component in svelte kit you will get message:
> svelte-paginate is incorrectly packaged. Please contact the package author to fix.

<img width="532" alt="image" src="https://user-images.githubusercontent.com/1398808/146274697-2bfce85a-05af-4bbe-aca1-4970d19413b6.png">

Probably it should be migrated to new packaging system presented in kit - https://kit.svelte.dev/docs#packaging

**Comment by zhuganglie** [comment]
Same problem here.

`
svelte-paginate doesn't appear to be written in CJS, but also doesn't appear to be a valid ES module (i.e. it doesn't have "type": "module" or an .mjs extension for the entry point). Please contact the package author to fix.

**Comment by bas-baskara** [comment]
> Same problem here.
> 
> ```
> svelte-paginate doesn't appear to be written in CJS, but also doesn't appear to be a valid ES module (i.e. it doesn't have "type": "module" or an .mjs extension for the entry point). Please contact the package author to fix.
> ```

I solved this simply by adding "type":"module" in package.json of svelte-paginate in node_modules. I have made PR for this at #27

**Comment by jarmitage** [comment]
@bas-baskara adding `"type":"module"` was helpful, thanks

I also had to add `"svelte": "src/index.js"` as well to address this error:

```
Unknown file extension ".svelte" for /repo/node_modules/svelte-paginate/src/PaginationNav.svelte
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".svelte" for /repo/node_modules/svelte-paginate/src/PaginationNav.svelte
    at new NodeError (node:internal/errors:372:5)
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:76:11)
    at defaultGetFormat (node:internal/modules/esm/get_format:118:38)
    at defaultLoad (node:internal/modules/esm/load:21:20)
    at ESMLoader.load (node:internal/modules/esm/loader:407:26)
    at ESMLoader.moduleProvider (node:internal/modules/esm/loader:326:22)
    at new ModuleJob (node:internal/modules/esm/module_job:66:26)
    at ESMLoader.#createModuleJob (node:internal/modules/esm/loader:345:17)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:304:34)
    at async ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:82:21)
```

**Comment by bas-baskara** [comment]
@jarmitage Yes thank you, I added your suggestion and also typescript support into my PR because I need it in my project too.

**Comment by TahaSh** [comment]
Fixed in `v0.1.0`.
